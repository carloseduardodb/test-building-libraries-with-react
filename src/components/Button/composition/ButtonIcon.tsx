import React from 'react';

interface Icons {
  icon: React.ElementType;
}

export const ButtonIcon = ({ icon: Icon }: Icons) => (
  <div>
    <Icon />
  </div>
);
