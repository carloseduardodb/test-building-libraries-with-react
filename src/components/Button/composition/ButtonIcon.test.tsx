import renderer from 'react-test-renderer';
import React from 'react';
import { ButtonIcon } from './ButtonIcon';

describe('ButtonIcon', () => {
  it('renders correctly with a given icon', () => {
    const MockIcon = () => <svg data-testid="mock-icon" />;
    const tree = renderer.create(<ButtonIcon icon={MockIcon} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without an icon', () => {
    const MockIcon = () => <svg data-testid="mock-icon" />;
    const tree = renderer.create(<ButtonIcon icon={MockIcon} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
