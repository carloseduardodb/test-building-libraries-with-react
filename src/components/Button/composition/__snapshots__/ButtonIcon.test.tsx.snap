// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonIcon renders correctly with a given icon 1`] = `
<div>
  <svg
    data-testid="mock-icon"
  />
</div>
`;

exports[`ButtonIcon renders correctly without an icon 1`] = `
<div>
  <svg
    data-testid="mock-icon"
  />
</div>
`;
